x/auth/ante/sigverify.go

```diff
@@ -237,7 +253,9 @@ func (svd SigVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 			err := authsigning.VerifySignature(pubKey, signerData, sig.Data, svd.signModeHandler, tx)
 			if err != nil {
 				var errMsg string
-				if sig.SkipSequenceCheck {
+				if OnlyLegacyAminoSigners(sig.Data) {
```

Comment:
in line 230 we already evaluated `OnlyLegacyAminoSigners(sig.Data)`. So let's store the result above and use it in both `if` statements.

Reply:
Shouldn't we just fail a transaction that contains mixed Sign modes?

Reply:
Depending on the users. This would solve my other [concern](https://github.com/cosmos/cosmos-sdk/pull/7285#discussion_r488619587).

Reply:
Mixed sign modes I believe is a valid use case that we need to support (see ADR020, and this [issue thread](https://github.com/cosmos/cosmos-sdk/issues/6078#issuecomment-628026933) on protobuf tx signing). When dealing with multisig transactions, each signer may be using their own independent sign mode, right?